pkg/git/git_keychain.go

```diff
@@ -1,26 +1,60 @@
 package git
 
 import (
-	"net"
+	"net/url"
+	"regexp"
 	"sort"
 	"strings"
 
-	"github.com/go-git/go-git/v5/plumbing/transport"
-	"github.com/go-git/go-git/v5/plumbing/transport/http"
-	gitSsh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	git2go "github.com/libgit2/git2go/v31"
 	"github.com/pkg/errors"
-	"golang.org/x/crypto/ssh"
 
 	"github.com/pivotal/kpack/pkg/secret"
+	"golang.org/x/crypto/ssh"
 )
 
+type Git2GoCredential interface {
+	Cred() (*git2go.Credential, error)
+}
+
+func asCredentialCallback(gitKeychain GitKeychain) git2go.CredentialsCallback {
+	return func(url string, username_from_url string, allowed_types git2go.CredentialType) (*git2go.Credential, error) {
```

Comment:
underscores?

Reply:
The underlying git2go implementation uses underscores. Let me switch that over.